From 11610debf750f15c7a104db7315dcd7d69e282a8 Mon Sep 17 00:00:00 2001
From: Alejandro Enedino Hernandez Samaniego <alhe@linux.microsoft.com>
Date: Sat, 26 Feb 2022 01:52:26 +0000
Subject: [PATCH] Makefile: Fix non-portable sh check for plugins

Upstream-Status: Pending

We previously held a patch that used "=" for comparison, but when
that patch got upstreamed it was changed to "==" which is non-portable,
resulting in an error:

/bin/sh: 6: [: acipher: unexpected operator
/bin/sh: 6: [: plugins: unexpected operator
/bin/sh: 6: [: hello_world: unexpected operator
/bin/sh: 6: [: hotp: unexpected operator
/bin/sh: 6: [: aes: unexpected operator
/bin/sh: 6: [: random: unexpected operator
/bin/sh: 6: [: secure_storage: unexpected operator

if /bin/sh doesnt point to bash.

Which in turn causes our do_install task to fail since plugins arent
where we expect them to be.


Signed-off-by: Alejandro Enedino Hernandez Samaniego <alhe@linux.microsoft.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b3f16aa..9359d95 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ prepare-for-rootfs: examples
 			cp -p $$example/host/optee_example_$$example $(OUTPUT_DIR)/ca/; \
 		fi; \
 		cp -pr $$example/ta/*.ta $(OUTPUT_DIR)/ta/; \
-		if [ $$example == plugins ]; then \
+		if [ $$example = plugins ]; then \
 			cp -p plugins/syslog/*.plugin $(OUTPUT_DIR)/plugins/; \
 		fi; \
 	done
-- 
2.25.1

